Add additional linux platform checking

astares opened the pull request:
Fix #7802

astares committed:
Add additional linux platform checking

Fix #7802

astares commented:
CI failures unrelated

astares commented:
Any reviewer?

guillep commented:
I'm not particularly against it... But it has a problem: why stop with these distros? :)
Somebody will always complain their distro is not in the list.

Maybe a solution more like linux's `uname` that returns the name of the current distro is more flexible from that point of view.

thoughts?

noha commented:
No, uname is not more flexible. There is is something called Linux standard base (https://en.wikipedia.org/wiki/Linux_Standard_Base) that defines a common system for major linux distros. The canonical way of getting information about an LSB system is 

`lsb_release -a`

But this is a binary. So relying on /etc/os-release is the proper way to do. But I'm with Guille here that making individual linux distros a mention in the image is not such a good idea. Neither is doing a substring search over the file ;)
If we define our linux target to be a LSB linux target which is feasible then I would make it detect linux and the Linux distro object offers access to the dictionary in the os-release file. In my system it looks like 

cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

which should be easy to parse and put into a dictionary

astares committed:
Merge branch 'Pharo9.0' of https://github.com/pharo-project/pharo into 7802-Add-additional-linux-platform-checking

astares commented:
@guillep 
uname can only be applied to general Unix. It gives you "Linux" on Linux.  The "/etc/os-release" solution and the LDB approach is the way forward in Linux world

@noha 
OK - I reworked and generalized the PR:
 - removed the specific #isXXXX checking 
 - parse the /etc/os-release to have access to all properties 
 - added better utility methods to check for the distribution and Linux derivate following the Linux specs

astares committed:
- do not list specfic distros
- parse the /etc/os-release to have access to ID and ID_LIKE following https://www.freedesktop.org/software/systemd/man/os-release.html

guillep commented:
I think @noha kind of understood what I explained poorly :)
What I meant is that I'd prefer a method `distributionName` rather than 10 `is[Distro]` methods

astares commented:
@guillep Yes - and thats what and why I wrote: the isXXX methods now got removed (and I included the other changes suggested too)

So please review again how the PR looks now so we can move on. Thanks in advance!

noha commented on src/System-Platforms/UnixPlatform.class.st:
```diff
@@ -106,6 +103,21 @@ UnixPlatform >> lineEnding [
+		props at: (each copyUpTo: $=) asSymbol put: (each copyAfter: $=) ]].
+
+	"Do nothing when it does not exist"
+	[ extractor value ] on: FileDoesNotExistException do: [ self error: 'Only available on modern distributions' ].
```
Can we replace "modern distributions" with "distributions supporting the linux standard base" ?

astares replied:
@noha  Yes - we can ;)

I changed the comment

astares committed:
change comment to point to LSB
